remote/coding/coding.go

```diff
@@ -0,0 +1,334 @@
+package coding
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/drone/drone/model"
+	"github.com/drone/drone/remote"
+	"github.com/drone/drone/remote/coding/internal"
+	"github.com/drone/drone/shared/httputil"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultURL = "https://coding.net" // Default Coding URL
+)
+
+// Opts defines configuration options.
+type Opts struct {
+	URL        string   // Coding server url.
+	Client     string   // Coding oauth client id.
+	Secret     string   // Coding oauth client secret.
+	Scopes     []string // Coding oauth scopes.
+	Machine    string   // Optional machine name.
+	Username   string   // Optional machine account username.
+	Password   string   // Optional machine account password.
+	SkipVerify bool     // Skip ssl verification.
+}
+
+// New returns a Remote implementation that integrates with a Coding Platform or
+// Coding Enterprise version control hosting provider.
+func New(opts Opts) (remote.Remote, error) {
+	remote := &Coding{
+		URL:        defaultURL,
+		Client:     opts.Client,
+		Secret:     opts.Secret,
+		Scopes:     opts.Scopes,
+		Machine:    opts.Machine,
+		Username:   opts.Username,
+		Password:   opts.Password,
+		SkipVerify: opts.SkipVerify,
+	}
+	if opts.URL != defaultURL {
+		remote.URL = strings.TrimSuffix(opts.URL, "/")
+	}
+
```

Comment:
@bradrydzewski But we have enterprise edition using another hostname. And we have self-host edition which may have even different hostnames.